### PR TITLE
AMDGPU: Add baseline tests for flat-may-alias private atomic expansions

### PR DESCRIPTION
**Pull request by arsenm**

### CODE REVIEW

**Comment by arsenm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109406?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#109406** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109406?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#102599** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102599?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#102461** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102461?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#102462](https://github.com/llvm/llvm-project/pull/102462) <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102462?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#103938](https://github.com/llvm/llvm-project/pull/103938) <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/103938?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/test/CodeGen/AMDGPU/flat_atomics_i64_noprivate.ll [hunk 1]
```diff
@@ -0,0 +1,6911 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc -mtriple=amdgcn -mcpu=bonaire < %s | FileCheck -check-prefix=GCN1 %s
```
Comment (rampitec) [inline llvm/test/CodeGen/AMDGPU/flat_atomics_i64_noprivate.ll:2]: Why GCN1 and GCN2? GFX7 and GFX8 are easier to understand.

Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/flat_atomics_i64_noprivate.ll:2]: Copied from the other test

**Comment by arsenm** [comment]
ping

**Review by rampitec — APPROVED** [review]

**Pull request merged**